### ISSUE TITLE
CRITICAL: istio-initializer needs to ignore hostNetwork pod specs

### ISSUE THREAD

**Issue by dcberg** [istio/istio.github.io#655]
The istio-initializer should never try to inject a sidecar into a pod spec that has `hostNetwork: true`. The problem is that the iptable changes are assumed to be within the pod when, in fact, they are changing the routing at the host level. This often results in routing failures within a node which can affect the network provider within the cluster causing additional pod failures.

**Comment by linsun** [comment]
Initial thought is to add logic to check if the object has `hostNetwork: true` in here:  https://github.com/istio/pilot/blob/master/platform/kube/inject/inject.go#L445. cc @ayj

**Comment by ayj** [comment]
Yes, that looks right, e.g.

```diff
diff --git a/platform/kube/inject/inject.go b/platform/kube/inject/inject.go
index 748b0f5..cf081a0 100644
--- a/platform/kube/inject/inject.go
+++ b/platform/kube/inject/inject.go
@@ -479,6 +479,16 @@ func intoObject(c *Config, in interface{}) (interface{}, error) {
        templateObjectMeta := templateValue.FieldByName("ObjectMeta").Addr().Interface().(*metav1.ObjectMeta)
        templatePodSpec := templateValue.FieldByName("Spec").Addr().Interface().(*v1.PodSpec)
 
+       // Skip injection when host networking is enabled. The problem is
+       // that the iptable changes are assumed to be within the pod when,
+       // in fact, they are changing the routing at the host level. This
+       // often results in routing failures within a node which can
+       // affect the network provider within the cluster causing
+       // additional pod failures.
+       if templatePodSpec.HostNetwork {
+               return out, nil
+       }
+
        for _, m := range []*metav1.ObjectMeta{objectMeta, templateObjectMeta} {
                if m.Annotations == nil {
                        m.Annotations = make(map[string]string)
```

**Comment by linsun** [comment]
@dcberg, thanks for reporting this.  This issue has been fixed in Pilot master.

**Comment by linsun** [comment]
@dcberg this issue has been fixed in both pilot branch release-0.2 and master.  You can pick it up at the 0.2.8 release when it is avail.
